src/solidity-frontend/solidity_convert.cpp

```diff
@@ -2578,6 +2777,32 @@ bool solidity_convertert::get_constructor_call(
 
   const nlohmann::json constructor_ref = find_decl_ref(ref_decl_id);
 
+  // check for implicit constructor
+  if(constructor_ref.empty())
+  {
+    std::string name, id;
+    if(get_contract_name(ref_decl_id, name))
+      return true;
+    id = name;
+    if(context.find_symbol(id) == nullptr)
+      return true;
+
+    symbolt s = *context.find_symbol(id);
+    typet type = s.type;
+
+    new_expr = exprt("symbol", type);
```

Comment:
could you add some comments to explain what we are trying to make here? something like: 
https://github.com/esbmc/esbmc/blob/b21b903debd9569f41eadf0d4be722920e087b4b/src/clang-cpp-frontend/clang_cpp_adjust_code.cpp#L162-L164

Reply:
Sure, I will add some comments.